app/src/main/java/com/google/android/samples/socialite/ui/player/preloadmanager/PreloadManagerWrapper.kt

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (C) 2024 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.samples.socialite.ui.player.preloadmanager
+
+import android.content.Context
+import android.os.Looper
+import androidx.annotation.MainThread
+import androidx.media3.common.MediaItem
+import androidx.media3.exoplayer.DefaultLoadControl
+import androidx.media3.exoplayer.DefaultRendererCapabilitiesList
+import androidx.media3.exoplayer.DefaultRenderersFactory
+import androidx.media3.exoplayer.source.DefaultMediaSourceFactory
+import androidx.media3.exoplayer.source.preload.DefaultPreloadManager
+import androidx.media3.exoplayer.source.preload.DefaultPreloadManager.Status.STAGE_LOADED_TO_POSITION_MS
+import androidx.media3.exoplayer.source.preload.TargetPreloadStatusControl
+import androidx.media3.exoplayer.trackselection.DefaultTrackSelector
+import androidx.media3.exoplayer.upstream.DefaultBandwidthMeter
+import com.google.android.samples.socialite.ui.home.timeline.TimelineMediaItem
+import com.google.android.samples.socialite.ui.home.timeline.TimelineMediaType
+
+/**
+ * Created by Mayuri Khinvasara on 12/08/24.
+ * Wrapper class to manage all functionalities of preload manager of exoplayer especially for short form content
+ */
+@androidx.media3.common.util.UnstableApi
+class PreloadManagerWrapper
+private constructor(
+    private val defaultPreloadManager: DefaultPreloadManager,
+) {
+    private val preloadWindow: ArrayDeque<Pair<MediaItem, Int>> = ArrayDeque()
+
+    // Default window size for preload manager
+    private var preloadWindowSize = 5
```

Comment:
can we add a bit more explanation about what 'window size' means here? as someone who doesn't understand the API, i'm not sure what window size means or how to set the window size for my app

Reply:
Ack. will add